cmd/codegen/hack/generate_and_sort_crds.sh

```diff
@@ -45,13 +41,8 @@ if ! $CONTROLLERGEN --version | grep -q "${CONTROLLERGEN_VERSION}" ; then
 fi
 
 # Run controller-gen
-CONTROLLERGEN_CRDS_YAML="${tmpdir}/crds-controllergen.yaml"
-${CONTROLLERGEN} crd webhook paths="./pkg/apis/..." output:stdout > "${CONTROLLERGEN_CRDS_YAML}"
-
-# Patch existing CRDs with the descriptions from controller-gen
-PATCHED_CRDS_YAML="${tmpdir}/crds-patched.yaml"
-COPY_FILE="${CONTROLLERGEN_CRDS_YAML}" run_yq -f cmd/codegen/hack/patch_crd_descriptions.jq "${CRDS_YAML}" "${CONTROLLERGEN_CRDS_YAML}" \
-  --slurp --sort-keys --explicit-start --yaml-output > "${PATCHED_CRDS_YAML}"
-
-# Override previous CRDs file
-cp "${PATCHED_CRDS_YAML}" "${CRDS_YAML}"
+${CONTROLLERGEN} crd webhook paths="./pkg/apis/..." output:stdout > $CRDS_YAML
+# Sort
+run_yq  --slurp --sort-keys --explicit-start --yaml-output -i 'sort_by(.metadata.name)[]' $CRDS_YAML
+# Set spec.preserveUnknownFields=false since it can't be done with controller-gen https://github.com/kubernetes-sigs/controller-tools/issues/476
+run_yq -yi '.spec.preserveUnknownFields=false' $CRDS_YAML
```

Comment:
According to [these docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning), this is the default for `apiextensions.k8s.io/v1`? Maybe it's not needed anymore and safe to just remove it?

Also, this could be done in a single call to `yq`

Reply:
you're right. Removed in https://github.com/rancher/fleet/pull/1904/commits/6a8c0e141caf98e8a9d344141c4c29d07b413033